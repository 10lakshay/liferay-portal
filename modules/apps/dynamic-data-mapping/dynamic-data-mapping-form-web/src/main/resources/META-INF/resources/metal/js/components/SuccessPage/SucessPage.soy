{namespace SuccessPage}

{template .render}
	{@param? contentLabel: string}
	{@param? successPageSettings: ?}
	{@param? titleLabel: string}
	{@param? _handleSuccessPageSettingsChanged: any}

	{delcall SuccessPage.container}
		{param contentLabel: $contentLabel /}
		{param successPageSettings: $successPageSettings /}
		{param titleLabel: $titleLabel /}
		{param _handleSuccessPageSettingsChanged: $_handleSuccessPageSettingsChanged /}
	{/delcall}
{/template}

{deltemplate SuccessPage.container}
	{@param? contentLabel: string}
	{@param? successPageSettings: ?}
	{@param? titleLabel: string}
	{@param? _handleSuccessPageSettingsChanged: any}

	<div class="ddm-page-success-layout">
		<div class="form-builder-success-page pt-2 pb-3">
			<label class="control-label">{$titleLabel}</label>
			<input data-setting="title" data-onkeyup="{$_handleSuccessPageSettingsChanged}" class="form-builder-success-page-title form-control" value="{$successPageSettings.title.en_US}" type="text" />

			<br />

			<label class="control-label">{$contentLabel}</label>
			<input data-setting="body" data-onkeyup="{$_handleSuccessPageSettingsChanged}" class="form-builder-success-page-content form-control" value="{$successPageSettings.body.en_US}" type="text" />
		</div>
	</div>
{/deltemplate}	