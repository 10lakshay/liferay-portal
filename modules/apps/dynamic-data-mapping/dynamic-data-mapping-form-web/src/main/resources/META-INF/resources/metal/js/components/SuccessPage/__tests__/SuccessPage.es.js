import {JSXComponent} from 'metal-jsx';
import {dom as MetalTestUtil} from 'metal-dom';
import SucessPageSettings from './__mock__/mockSuccessPage.es';
import SuccessPage from '../SuccessPage.es';

let component;
let successPageSettings;

describe.only(
	'SuccessPage',
	() => {
		beforeEach(
			() => {
				successPageSettings = JSON.parse(JSON.stringify(SucessPageSettings));

				jest.useFakeTimers();
			}
		);

		afterEach(
			() => {
				if (component) {
					component.dispose();
				}

				successPageSettings = null;
			}
		);

		it(
			'should render the default layour',
			() => {
			
				component = new SuccessPage(
					{
						contentLabel: 'Content',
						titleLabel: 'Title',
						successPageSettings
					}
				);

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should emit success page changed when success page title is changed',
			() => {
				const newPageSettings = {
					...successPageSettings,
					enabled: true
				}
			
				component = new SuccessPage(
					{
						contentLabel: 'Content',
						titleLabel: 'Title',
						successPageSettings: newPageSettings
					}
				);
				const spy = jest.spyOn(component, 'emit');
				const titleNode = component.element.querySelector('input[data-setting="title"]');

				titleNode.value = "Some title";

				jest.runAllTimers();

				MetalTestUtil.triggerEvent(titleNode, 'keyup', {});

				expect(spy).toHaveBeenCalledWith('successPageChanged', expect.anything());
			}
		);

		it(
			'should emit success page changed when success page body is changed',
			() => {
				const newPageSettings = {
					...successPageSettings,
					enabled: true
				}
			
				component = new SuccessPage(
					{
						contentLabel: 'Content',
						titleLabel: 'Title',
						successPageSettings: newPageSettings
					}
				);
				const spy = jest.spyOn(component, 'emit');
				const titleNode = component.element.querySelector('input[data-setting="body"]');

				titleNode.value = "Some description";

				jest.runAllTimers();

				MetalTestUtil.triggerEvent(titleNode, 'keyup', {});

				expect(spy).toHaveBeenCalledWith('successPageChanged', expect.anything());
			}
		);
	}
);