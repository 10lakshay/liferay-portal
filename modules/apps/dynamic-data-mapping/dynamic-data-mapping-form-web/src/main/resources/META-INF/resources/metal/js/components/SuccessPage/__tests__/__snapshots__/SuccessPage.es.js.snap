// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPage should render the default layour 1`] = `
<div class="ddm-page-success-layout">
  <div class="form-builder-success-page pt-2 pb-3">
    <label class="control-label">Title</label>
    <input data-setting="title" class="form-builder-success-page-title form-control" value="" type="text">
    <br>
    <label class="control-label">Content</label>
    <input data-setting="body" class="form-builder-success-page-content form-control" value="" type="text">
  </div>
</div>
`;
